wcfsetup/install/files/lib/data/user/UserAction.class.php

```diff
@@ -828,4 +828,38 @@ public function validateSaveSocialNetworkPrivacySettings() {
 	public function saveSocialNetworkPrivacySettings() {
 		// does nothing
 	}
+	
+	/**
+	 * Validates the 'resendActivationMail' action.
+	 * @throws UserInputException
+	 * @since 3.2
+	 */
+	public function validateResendActivationMail() {
+		$this->readObjects();
+		
+		foreach ($this->objects as $object) {
+			if (!$object->activationCode) {
+				throw new UserInputException('objectIDs');
+			}
+		}
+	}
+	
+	/**
+	 * Triggers a new activation email.
+	 * @since 3.2
+	 */
+	public function resendActivationMail() {
+		foreach ($this->objects as $object) {
```

Comment:
A new `activationCode` should be generated for each user (similar to `RegisterNewActivationCodeForm`).